src/main/java/com/adobe/aem/compgenerator/web/TabBuilderServlet.java

```diff
@@ -0,0 +1,241 @@
+package com.adobe.aem.compgenerator.web;
+
+import com.adobe.aem.compgenerator.models.*;
+import com.adobe.aem.compgenerator.utils.CommonUtils;
+import com.adobe.aem.compgenerator.web.model.Message;
+import com.fasterxml.jackson.core.util.DefaultIndenter;
+import com.fasterxml.jackson.core.util.DefaultPrettyPrinter;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static com.adobe.aem.compgenerator.Constants.*;
+
+public class TabBuilderServlet extends HttpServlet {
+    private static final Logger LOG = LogManager.getLogger(TabBuilderServlet.class);
+
+    private void updateTabProperties(JsonNode reConfig, Tab tab) {
+        if (reConfig.has("label")) {
+            tab.setLabel(reConfig.get("label").asText());
+        }
+        if (reConfig.has("fields")) {
+            List<String> updatedFields = new ArrayList<>();
+            JsonNode val = reConfig.get("fields");
+            if (val.isArray()) {
+                for (JsonNode jsonNode : val) {
+                    String field = jsonNode.get("value").asText();
+                    updatedFields.add(field);
+                }
+            }
+            tab.setFields(updatedFields);
+        }
+    }
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+        resp.setContentType("application/json; charset=UTF-8");
+        PrintWriter writer = resp.getWriter();
+        ObjectMapper mapper = new ObjectMapper();
+        Message msg = new Message(true, "Tab builder getter");
+        writer.write(mapper.writeValueAsString(msg));
+        writer.close();
+    }
+
+    @Override
+    /*
+     * Handles updates to the Dialog Tab configurations
+     */
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
```

Comment:
Lots of copy/pasted code to handle MAIN, SHARED, and GLOBAL cases.  Im guessing a lot of this can be consolidated into functions that will drastically shorten the code and make it more maintainable (i.e. bug fixes wont have to be applied in multiple places).

Reply:
I would agree, yeah a lot of consolidation could be done. I'll see about DRY'ing it up more.